### ISSUE TITLE
endpoint healthcheck fails when replicating to second harbor vm with same external db

### ISSUE THREAD

**Issue by arjunprasad2143** [goharbor/harbor#13730]
We have setup harbor as a standalone server with external database enabled. 
harbor version 2.1.1
external db - pgsql 10.14 (HA enabled)
Everything is working as expected.

Now we wanted to move this to a High Availability setup. So 2 harbor vms, HA enabled external db, HA enabled external redis and  external s3 minio storage.
Now to sync the 2 harbor vms we wanted to enable the replication between harbor vms. So we added a replication rule and set up an endpoint for replication. When I test the connection with basic authentication, its is shown as OK. But after couple of minutes the password gets changed automatically as a result the healthcheck is also shown as failed.

We cannot figure out why the password gets changed. The db is same for both vms. 
Any help would be appreciated.

If you need any other details I can share.

**Comment by wy65701436** [comment]
do you mean the password of admin on the second harbor VM gets changed automatically? Harbor has no this kind of mechanism to change the admin password automatically.

**Comment by arjunprasad2143** [comment]
> do you mean the password of admin on the second harbor VM gets changed automatically? Harbor has no this kind of mechanism to change the admin password automatically.

No. The access secret set on the replication registry endpoint gets changed. When I test the connection from the UI, it shows connection tested successfully and healthy. But after a few minutes this endpoint becomes unhealthy and when we check the logs it says invalid credentials. 

Below are the logs from core.log
"Dec 10 10:12:51 172.30.0.1 core[1823753]: 2020-12-10T09:12:51Z [ERROR] [/replication/adapter/native/adapter.go:126]: failed to ping registry https://harbor1.xxxxxxx.com: http status code: 401, body: 

I am also attaching the screengrab of the connection being tested successfully
![endpoint](https://user-images.githubusercontent.com/16715595/101882537-51e49480-3b96-11eb-96bf-00c96aee1ab7.png)

**Comment by bitsf** [comment]
Do you mean you are using two harbors with same database and redis and minio, and sync the two harbors by replication ?

**Comment by bitsf** [comment]
If you really want to share the same db, you should also sync the harbor/common/config/ folder, especially make sure the `*_SECRET` and `*_KEY` same.

**Comment by arjunprasad2143** [comment]
> Do you mean you are using two harbors with same database and redis and minio, and sync the two harbors by replication ?

This is correct. I will test the recommended solution.
But the replication is not only between 2 harbor vms. 
We have another replication also setup which also fails. Anyway I will test and let you know.

**Comment by arjunprasad2143** [comment]
> If you really want to share the same db, you should also sync the harbor/common/config/ folder, especially make sure the `*_SECRET` and `*_KEY` same.
> And I don't think you need to do replication as you already use same db and minio.

Today I did the tests and it failed again

Steps followed.
1. Start harbor1 with external db(HA), external redis, s3 minio
2. Copied the folder harbor/common/config/  of harbor1 to harbor2.
3. Updated few configs like external url, certs etc for harbor2 in the config files
4. Started the service on harbor2
5. Checked the replication status of a registry(external registry) configured. The status has changed from healthy to unhealthy.

As mentioned, we didnt check for replication between 2 harbor vms as its not desired for the setup. 

I made sure the CORE_SECRET, JOBSERVICE_SECRET and CSRF_KEY was the same on both harbor vms (in core and registryctl). 
Is there any other SECRET of KEY that I should check?
